src/java.base/share/classes/java/lang/invoke/GenerateJLIClassesHelper.java

```diff
@@ -53,16 +55,23 @@
     private static final String SPECIES_RESOLVE = "[SPECIES_RESOLVE]";
 
     static void traceLambdaForm(String name, MethodType type, Class<?> holder, MemberName resolvedMember) {
-        if (TRACE_RESOLVE) {
-            System.out.println(LF_RESOLVE + " " + holder.getName() + " " + name + " " +
-                    shortenSignature(basicTypeSignature(type)) +
-                    (resolvedMember != null ? " (success)" : " (fail)"));
+        if (TRACE_RESOLVE || CDS.isDumpLoadedClassList()) {
+            String traceLF = LF_RESOLVE + " " + holder.getName() + " " + name + " " +
+                    shortenSignature(basicTypeSignature(type));
+            if (TRACE_RESOLVE) {
+                System.out.println(traceLF + (resolvedMember != null ? " (success)" : " (fail)"));
+            }
```

Comment:
I suggest not to change the existing code.  Instead, have `CDS::traceLambdaFormInvoker`
to take individual parameters `Class<?> holder, String name, String shortenSignature`
(rather than the formatted string).   Something like:

```
    if (CDS.isDumpLoadedClassList()) {
          CDS.traceLambdaFormInvoker(holder, name, shortenSignature(basicTypeSignature(type));
    }
```

This also gives flexibility to CDS to decide on what format to write to the class list (like this case, you drop the text "success/fail")

In addition, the conditional check on `CDS.isDumpLoadedClassList()` is hard to relate to why CDS traces these events.   I see Ioi's comment on this method name too.   I agree with Ioi that `isDumpingClassList` makes more sense.

Reply:
Done